src/components/NewMessage.jsx

```diff
@@ -16,6 +18,7 @@ export default function NewMessage({ newMessage, sendMessage, setNewMessage }) {
             flex: 1,
           }}
         ></TextField>
+        <FileUploader onUploadComplete={onUploadComplete} />
```

Comment:
Why do we have separate upload and send buttons?

Reply:
They should be the same

Reply:
Please disable the send button once you are processing upload, sending the text, etc. to avoid user from clicking it multiple times and generating the same message multiple times.